### PR TITLE
feat: add fields for kroma deposit tx

### PR DESCRIPTION
**Pull request by fakedev9999**
### Description

This PR adds fields for Kroma deposit transactions.

Features for enabling tx circuit to handle Kroma deposit transactions will be added soon.

This PR is related to the [PR in ethers-rs repo](https://github.com/kroma-network/ethers-rs/pull/1).

Since the PR in ethers-rs repo is not merged yet, you need to make the below patch to Cargo.toml.

```
diff --git a/Cargo.toml b/Cargo.toml
index 2afd9e1d..46936f2c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -29,9 +29,9 @@ base64 = "0.13.0"
 ctor = "0.1"
 env_logger = "0.10"
 ethers = { version = "=2.0.7", features = ["ethers-solc"] }
-ethers-core = { git = "https://github.com/kroma-network/ethers-rs.git", branch = "feat/upstream-scroll-v0.10.0rc3", features = ["scroll"] }
+ethers-core = { git = "https://github.com/kroma-network/ethers-rs.git", branch = "feat/add-kroma-features", features = ["kroma"] }
 ethers-providers = "=2.0.7"
-ethers-signers = "=2.0.7"
+ethers-signers = { git = "https://github.com/kroma-network/ethers-rs.git", branch = "feat/add-kroma-features"}
 ff = "0.13"
 halo2_proofs = { git = "https://github.com/kroma-network/halo2.git", branch = "feat/upstream-scroll-v0.10.0rc3" }
 hash-circuit = { package = "poseidon-circuit", git = "https://github.com/kroma-network/poseidon-circuit.git", branch = "feat/upstream-scroll-v0.10.0rc3" }
```

### CODE REVIEW

**Review by chokobole — APPROVED** [review]
LGTM

File: eth-types/src/l2_types.rs [hunk 1]
```diff
@@ -169,6 +175,10 @@ impl TransactionTrace {
             v: self.v,
             r: self.r,
             s: self.s,
+            #[cfg(feature = "kroma")]
+            source_hash: Some(self.source_hash),
```
Comment (ashjeong) [inline eth-types/src/l2_types.rs:179]: Though I would prefer if the ordering of the `mint`/`source hash` could be standardized (aka `mint` before `source hash`, both after `s`) I do see that the order of other fields is also not standardized. Besides this comment, LGTM

**Review by ashjeong — APPROVED** [review]

File: eth-types/Cargo.toml [hunk 2]
```diff
@@ -29,9 +29,10 @@ base64.workspace = true
 default = ["warn-unimplemented"]
 warn-unimplemented = []
 shanghai = []
+kroma = []
```
Comment (dongchangYoo) [inline eth-types/Cargo.toml:32]: we shoud give `kroma` option to ethers-rs

File: eth-types/src/l2_types.rs (hunk 1 above)
Comment (fakedev9999) [reply eth-types/src/l2_types.rs:179]: I tried to follow the order of our l2 transaction trace, with minimum diff. An example of our l2 transaction trace is as below.
```
{
            "type": 2,
            "nonce": 1,
            "txHash": "0x993256bed459a2ebf390e4b7c1ab915ef8b48637792f6601297d9cc1423cc21d",
            "gas": 100000,
            "gasPrice": "0xb390d1d4",
            "from": "0x3c44cdddb6a900fa2b585dd299e03d12fa4293bc",
            "to": "0x9fe46736679d2d9a65f0992f2272de9f3c7fa6e0",
            "chainId": "0x385",
            "mint": null,
            "value": "0x0",
            "data": "0xa9059cbb00000000000000000000000090f79bf6eb2c4f870365e785982e1f101e93b9060000000000000000000000000000000000000000000000000400000000000000",
            "isCreate": false,
            "sourceHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
            "v": "0x1",
            "r": "0xdaddb750ce136c22a007a3365b3d7c41535f87f95ce54a29b929e287838ac190",
            "s": "0x2ac416e1d81fc770735cf96d2f396acccbcb4fe69de27384b36ea7696a0e98d1",
            "accessList": [
                {
                    "address": "0x9fe46736679d2d9a65f0992f2272de9f3c7fa6e0",
                    "storageKeys": [
                        "0x215be5d23550ceb1beff54fb579a765903ba2ccc85b6f79bcf9bda4e8cb86034"
                    ]
                }
            ]
        },
```

File: eth-types/Cargo.toml (hunk 2 above)
Comment (fakedev9999) [reply eth-types/Cargo.toml:32]: That's done in Cargo.toml!
```
ethers = { version = "=2.0.7", features = ["ethers-solc", "kroma"] }
ethers-core = { git = "https://github.com/kroma-network/ethers-rs.git", branch = "feat/upstream-scroll-v0.10.0rc3", features = ["kroma"] }
ethers-providers = { git = "https://github.com/kroma-network/ethers-rs.git", branch = "feat/upstream-scroll-v0.10.0rc3", features = ["kroma"] }
ethers-signers = { git = "https://github.com/kroma-network/ethers-rs.git", branch = "feat/upstream-scroll-v0.10.0rc3"}
```

**Comment by dongchangYoo** [comment]
Scroll upstream work is paused, so this PR will be merged later.

**Pull request closed without merging**